kable-core/src/appleMain/kotlin/bluetooth/CheckBluetoothIsOn.kt

```diff
@@ -0,0 +1,37 @@
+package com.juul.kable.bluetooth
+
+import com.juul.kable.CentralManager
+import com.juul.kable.InternalException
+import com.juul.kable.UnmetRequirementException
+import com.juul.kable.UnmetRequirementReason.BluetoothDisabled
+import platform.CoreBluetooth.CBManagerState
+import platform.CoreBluetooth.CBManagerStatePoweredOff
+import platform.CoreBluetooth.CBManagerStatePoweredOn
+import platform.CoreBluetooth.CBManagerStateResetting
+import platform.CoreBluetooth.CBManagerStateUnauthorized
+import platform.CoreBluetooth.CBManagerStateUnknown
+import platform.CoreBluetooth.CBManagerStateUnsupported
+
+/**
+ * @throws UnmetRequirementException If [CentralManager] state is not [CBManagerStatePoweredOn].
+ */
+internal fun CentralManager.checkBluetoothIsOn() {
```

Comment:
Not sure if this file would be the right place, but it would be really useful to have a flow of the CBManagerState itself. Is that factored into your plan somewhere?

Reply:
I've decided that Kable should not provide bluetooth state for consumers — it is too difficult to provide a unified API.

I'm trying to drive Kable in a direction where the entire API can be used from `common` source sets (write once, use everywhere paradigm).

In this particular instance: `CBManager*` states encompass:
- is BLE supported or not
- is BLE powered on or not
- does user have **permission** to use BLE

Whereas Android has all three of the above bullets available independently (and there is no live-updating permission info available on Android -- it is on-demand).

If `CBManager*` states didn't include the permission aspect, then it is would trivial to provide a unified API (of a flow of "is BLE powered on or not"), but since Apple doesn't separate those states, means that I just can't provide such an API.

So, we're left with only being able to provide an on-demand approach. Which sadly, I may even need to abandon this; as I've found that my findings of being able to suppress the dialog when using BLE isn't actually possible on Apple (I was previously mistaken on my findings: I am able to suppress for BLE on/off dialog, but permissions always comes up).

Making this API super intrusive...ultimately, it is hot garbage that Core Bluetooth doesn't let you query any information about BLE availability without showing the user a dialog. 😢 

I haven't completely abandoned the idea, but a unified API for these things may be impossible; really just trying to find the best compromise (and still have a reasonably friendly API).

Reply:
Additional details can also be found here: https://github.com/JuulLabs/kable/issues/737